WebApp/src/main/java/com/isa/webapp/controller/TeacherController.java

```diff
@@ -1,115 +1,80 @@
 package com.isa.webapp.controller;
 
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.isa.webapp.model.GradeForm;
-import com.isa.webapp.model.Subjects;
-import com.isa.webapp.model.User;
-import com.isa.webapp.model.UserRole;
-import com.isa.webapp.service.StudentService;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.isa.webapp.model.*;
+import com.isa.webapp.repository.GradeRepository;
+import com.isa.webapp.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
 import org.springframework.security.core.annotation.AuthenticationPrincipal;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.*;
 
 import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.ArrayList;
+import java.time.LocalDate;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 
 @Controller
+@RequiredArgsConstructor
 public class TeacherController {
 
-    private final StudentService studentService;
+    private final UserRepository userRepository;
 
-    @Autowired
-    public TeacherController(StudentService studentService) {
-        this.studentService = studentService;
-    }
-
-    private static final String USERS_JSON_FILE = "users.json";
+    private final GradeRepository gradeRepository;
 
     @GetMapping("/teacher/students")
     public String showStudentList(Model model, @AuthenticationPrincipal UserDetails userDetails) throws IOException {
-        boolean isTeacher = userDetails.getAuthorities().stream()
-                .anyMatch(a -> a.getAuthority().equals("TEACHER"));
+        boolean isTeacher = userDetails.getAuthorities().stream().anyMatch(a -> a.getAuthority().equals("TEACHER"));
         User user = (User) userDetails;
-        List<User> students = getStudentsForTeacher();
+        /*        List<User> students = getStudentsForTeacher();*/
+        List<User> students = userRepository.findAllByUserRole(UserRole.STUDENT);
         model.addAttribute("students", students);
         model.addAttribute("isTeacher", isTeacher);
         model.addAttribute("username", user.getFirstName() + " " + user.getLastName());
```

Comment:
właściwie przy tych atrybutach zamiast zmiennych itp to bym dał same wywołania funkcji prywatnych, które mogą być np. w tym serwisie, który wypadałoby stworzyć zamiast korzystać bezpośrednio na repo